pkg/controllers/dynakube/activegate/internal/tls/reconciler.go

```diff
@@ -0,0 +1,171 @@
+package tls
+
+import (
+	"context"
+	"crypto/x509"
+	"net"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/consts"
+	"github.com/Dynatrace/dynatrace-operator/pkg/logd"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/certificates"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/conditions"
+	k8slabels "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
+	k8ssecret "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/secret"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/timeprovider"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	activeGateSelfSignedTLSCommonNameSuffix = "-activegate.dynatrace"
+
+	tlsCrtDataName = "server.crt"
+)
+
+var (
+	log = logd.Get().WithName("dynakube-activegate-tls-secret")
+)
+
+type Reconciler struct {
+	client       client.Client
+	apiReader    client.Reader
+	dk           *dynakube.DynaKube
+	timeProvider *timeprovider.Provider
+}
+
+type ReconcilerBuilder func(client client.Client, apiReader client.Reader, dk *dynakube.DynaKube) *Reconciler
+
+func NewReconciler(client client.Client, apiReader client.Reader, dk *dynakube.DynaKube) *Reconciler {
+	return &Reconciler{
+		client:       client,
+		dk:           dk,
+		apiReader:    apiReader,
+		timeProvider: timeprovider.New(),
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context) error {
+	if r.dk.ActiveGate().IsEnabled() && r.dk.ActiveGate().IsAutomaticTlsSecretEnabled() {
```

Comment:
```suggestion
	if r.dk.ActiveGate().IsEnabled() && r.dk.ActiveGate().IsAutomaticTlsSecretEnabled() {
```
naming a bit confusing, we use IsEnabled, and then IssomethingEnabled, I prefer to stay with SomethingIsEnabled

wdyt?

Reply:
I think we don't want to duplicate ActiveGate name in condition `r.dk.ActiveGate().IsActiveGateEnabled()`.

But regarding AG feature I prefer `IsAutomaticTlsSecretEnabled` (Is<feature>Enabled) 
```
func (ag *Spec) IsRoutingEnabled() bool {
func (ag *Spec) IsApiEnabled() bool {
func (ag *Spec) IsMetricsIngestEnabled() bool {
func (ag *Spec) IsAutomaticTlsSecretEnabled() bool {
```